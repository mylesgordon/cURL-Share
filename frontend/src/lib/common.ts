export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}