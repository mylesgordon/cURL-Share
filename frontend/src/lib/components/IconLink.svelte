<script lang="ts">
	export let description: string;
	export let href: string;
</script>

<a {href} aria-label={description}>
	<slot />
</a>
