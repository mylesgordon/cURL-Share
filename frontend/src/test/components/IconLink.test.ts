import { render, screen } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';
import IconLink from '$lib/components/IconLink.svelte';
import type { ComponentProps } from 'svelte';

describe('IconLink component', () => {
	const props: ComponentProps<IconLink> = {
		description: 'Settings button',
		href: 'https://example.com'
	};
	testMatchingSnapshot(IconLink, props);

	it('should set href and description correctly', () => {
		render(IconLink, props);
		const link = screen.getByLabelText(props.description);
		expect(link.getAttribute('href')).toBe(props.href);
	});
});
