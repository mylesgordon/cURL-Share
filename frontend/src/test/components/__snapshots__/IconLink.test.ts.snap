// Vitest Snapshot v1

exports[`IconLink component > should match snapshot 1`] = `
<body>
  <div>
    <a
      aria-label="Settings button"
      href="https://example.com"
    />
    <!--&lt;IconLink&gt;-->
  </div>
</body>
`;
